src/util/thread.h

```diff
@@ -26,6 +26,7 @@ with this program; if not, write to the Free Software Foundation, Inc.,
 #include "../threading/mutex_auto_lock.h"
 #include "porting.h"
 #include "log.h"
+#include "container.h" // MutexedQueue
```

Comment:
Do we really intend to postpone header inclusions for classes required by templated classes like that? Well ok...